test/test_auth.py

```diff
@@ -201,3 +201,117 @@ def test_scram_both(bouncer):
 
     # SCRAM password in userlist.txt
     bouncer.test(dbname="p62", user="scramuser1", password="foo")
+
+
+@pytest.mark.skipif("WINDOWS", reason="Windows does not have SIGHUP")
+def test_connection_with_missing_stats_user_and_enabled_auth_query_without_global_auth_dbname(
+    bouncer,
+):
+    """
+    Regression for https://github.com/pgbouncer/pgbouncer/issues/314
+
+    Check that pgbouncer is not crashing on connection to the virtual pgbouncer virtual db when user is missing in
+    userlists.txt and auth_query enabled.
+
+    It covers case when global auth_dbname does not set and target database used for auth_dbname in auth query
```

Comment:
Suggested description:

It covers `the` case when auth_dbname is not configured and the target database is used as the authentication database.

Reply:
I assigned simpler test function names and added additional information in the doc-string